Apps/DiceRoller/DiceRoller/ISDRProjectManager.m

```diff
@@ -24,15 +24,77 @@ - (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
-    int curProjID = [dm getProjectID];
-    NSString *curProjIDStr = (curProjID > 0) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
     
-    [projectLbl setText:[NSString stringWithFormat:@"Uploading to Project: %@", curProjIDStr]];
+    pm = [ [ISDRProjectManager alloc] init];
+    
+    bool projectValid = [pm projectHasValidFields];
+    
+    if( projectValid ){
+        
+        int curProjID = [dm getProjectID];
+        NSString *curProjIDStr = (curProjID > 100) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
+        [projectLbl setText:[NSString stringWithFormat:@"Uploading to Project: %@", curProjIDStr]];
+        
+    }else{
+        
+        // If we are in here it means that the project selected does not have properly formatted fields.
+        // Present an error message to the user and set project back to default project.
+        
+        [self.view makeWaffle:@"Error: Selected Project did not have properly formatted fields. " duration:WAFFLE_LENGTH_LONG position:WAFFLE_BOTTOM image:WAFFLE_RED_X];
+        int curProjID = 876;
+        [dm setProjectID:curProjID];
+        NSString *curProjIDStr = (curProjID > 100) ? [NSString stringWithFormat:@"%d", curProjID] : kNO_PROJECT;
+        [projectLbl setText:[NSString stringWithFormat:@"Uploading to Project: %@", curProjIDStr]];
+
+    }
+}
+
+
+- (bool) projectHasValidFields {
+    
+    //Checks whether a method has three fields named yellow, white and sum. If it the project does, then returns true. If not, returns false.
+    
+    bool hasWhiteDieField = false;
+    bool hasYellowDieField = false;
+    bool hasSumField = false;
+    
+    //Declares an instance of API
+    api = [API getInstance];
+    
+    //Pulls down fields for the iSENSE project
+    NSArray *projectFields = [api getProjectFieldsWithId:[dm getProjectID]];
+    
+    //Loop through all the project fields to see if they are the required fields.
+    for (RProjectField *field in projectFields) {
+        
+        if ([field.name.lowercaseString rangeOfString:@"white"].location != NSNotFound) {
+            // if we're here, this means the project has a field for the white die.
+            hasWhiteDieField = true;
+        } else if ([field.name.lowercaseString rangeOfString:@"yellow"].location != NSNotFound) {
+            // if we're here, this means the project has a field for the yellow die.
+            hasYellowDieField = true;
+            
+        } else if ([field.name.lowercaseString rangeOfString:@"sum"].location != NSNotFound) {
+            // if we're here, this means the project has a field for the sum.
+            hasSumField = true;
+            
+        }
+    }
+
+    
+    //Check to see if all three booleans are true. If they are, the project is valid! If not the project is invalid.
+    if( (hasWhiteDieField == true) && (hasYellowDieField == true) && (hasSumField == true) ){
+        return true;
+    }
+    else{
+        return false;
+    }
```

Comment:
Code condensing:
Lines 92-97 ISDRProjectManager.m can be reduced to:
```return (hasWhiteDieField && hasYellowDieField && hasSumField)```
- No need to evaluate if the values are equivalent to true since they are boolean values.  Additionally, you can return the result of that if statement (i.e. true for true, false for false)